src/app/server-cluster/BUILD.gn

```diff
@@ -25,6 +25,7 @@ source_set("server-cluster") {
     "${chip_root}/src/app:attribute-access",
     "${chip_root}/src/app:command-handler-interface",
     "${chip_root}/src/app:paths",
+    "${chip_root}/src/app/util/persistence",
```

Comment:
This seems to tie things in to ember - we pull in persistence which pulls in util/types.

Is there a way to move things around so we limit ember dependencies? What does persistence need for util::types?

Reply:
SafeAttributePersistence seems to say `uses a different key space from AttributePersistenceProvider` so it is also not usable (because ember likely uses native endianness). 

It seems that the requirements are broader: we need a way to have a persistence provider both good for new code or AAI (i.e. write endianess-safe data) and for usage in backwards compatibility (i.e. what util/persistenceprovider does).

We may need to enhance the SafeAttributePersistenceProvider or have some derived class that provides the ember compatibility bits as well. Please propose some approaches, we may want to do slack/docs or the issue itself as this seems to be a bit more painful than expected.